File: VB/CircledProgressBarViewInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.Utils.Drawing
Imports DevExpress.LookAndFeel

Namespace CircledProgressBarEditor
	Friend Class CircledProgressBarViewInfo
		Inherits ProgressBarViewInfo
		' constructor
		Public Sub New(ByVal item As RepositoryItem)
			MyBase.New(item)
		End Sub

		Public Overrides ReadOnly Property ProgressPainter() As ProgressBarObjectPainter
			Get
				Return New CircledProgressBarObjectPainter(Owner)
			End Get
		End Property
	End Class
End Namespace

File: CS/CircledProgressBarViewInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Repository;
using DevExpress.Utils.Drawing;
using DevExpress.LookAndFeel;

namespace CircledProgressBarEditor
{
    class CircledProgressBarViewInfo : ProgressBarViewInfo
    {
        // constructor
        public CircledProgressBarViewInfo(RepositoryItem item) : base(item) { }

        public override ProgressBarObjectPainter ProgressPainter
        {
            get
            {
                return new CircledProgressBarObjectPainter(Owner);
            }
        }
    }
}
